src/cargo/ops/cargo_install.rs

```diff
@@ -651,11 +653,59 @@ pub fn install(
             true,
             current_rust_version.as_ref(),
         )?;
-        let mut installed_anything = true;
-        if let Some(installable_pkg) = installable_pkg {
-            installed_anything = installable_pkg.install_one()?;
+        if dry_run {
+            match installable_pkg {
+                Some(installable_pkg) => dry_run_report(
```

Comment:
See https://github.com/rust-lang/cargo/issues/11123#issuecomment-2004139099

Reply:
If this is indeed required, I'll see what I can do to rework this and go into the more fine-grained compilation steps to have a proper dry-run. It might take some time though, as it is almost starting from scratch.

@rustbot author